python_modules/libraries/dagster-databricks/dagster_databricks/components/databricks_asset_bundle/component.py

```diff
@@ -44,8 +57,19 @@ class DatabricksAssetBundleComponent(Component, Resolvable):
     def databricks_configs(self) -> DatabricksConfigs:
         return DatabricksConfigs(databricks_configs_path=self.databricks_configs_path)
 
-    def get_asset_spec(self) -> AssetSpec:
-        raise NotImplementedError()
+    def get_asset_spec(self, task: DatabricksBaseTask) -> AssetSpec:
+        return AssetSpec(
+            key=snake_case(task.task_key),
```

Comment:
are tasks generally named in relation to the asset they produce? just trying to figure out if there's a better default asset key here.

Reply:
separately, I've been warming to a pattern where the return type of this `get_asset_spec` method is actually `Optional[AssetSpec]` to make it easy to filter out certain things from being represented as specs. Thoughts on that here?

Reply:
> separately, I've been warming to a pattern where the return type of this get_asset_spec method is actually Optional[AssetSpec] to make it easy to filter out certain things from being represented as specs. Thoughts on that here?

So the logic excluding some assets would be in get_asset_spec? Should get_asset_spec return the asset spec as-is based on the data that was provided to the method?

Perhaps you have a concrete use case?